eox_tagging/models.py

```diff
@@ -47,35 +49,44 @@ def invalid(self):
         """Returns all invalid tags."""
         return self.exclude(invalidated_at=None)
 
-    def find_by_owner(self, owner):
+    def find_by_owner(self, owner_type, owner_id):
         """Returns all valid tags owned by owner_id."""
-        model = owner.__class__
-        ctype = ContentType.objects.get_for_model(model)
 
-        return self.valid().filter(
+        owner, ctype = self.__get_object_for_this_type(owner_type, owner_id)
+
+        return self.filter(
             owner_type=ctype,
             owner_object_id=owner.id,
         )
 
-    def find_all_tags_for(self, target_object):
-        """Returns all valid on an object."""
-        if get_model_name(target_object) in OPAQUE_KEY_PROXY_MODEL_TARGETS:
-            model = OpaqueKeyProxyModel
-            target_id = OpaqueKeyProxyModel.objects.get(opaque_key=target_object.course_id).id
-        else:
-            model = target_object.__class__
-            target_id = target_object.id
-        ctype = ContentType.objects.get_for_model(model)
+    def find_all_tags_for(self, target_type, target_id):
+        """Returns all valid tags on an object."""
+        if target_type in OPAQUE_KEY_PROXY_MODEL_TARGETS:
+            target_type = PROXY_MODEL_NAME
 
-        return self.valid().filter(
+        target, ctype = self.__get_object_for_this_type(target_type, target_id)
+
+        return self.filter(
             target_type=ctype,
-            target_object_id=target_id,
+            target_object_id=target.id,
         )
 
     def hard_delete(self):
         """ Method for deleting Tag objects"""
         return super(TagQuerySet, self).delete()
 
+    def __get_object_for_this_type(self, object_type, object_id):
+        """Function that returns the correct content type given a type."""
+        ctype = ContentType.objects.get(model=object_type)
+        if object_type == PROXY_MODEL_NAME:
+            object_id = object_id.get("course_id")
```

Comment:
if you apply something  similar to the suggestion number 6 in the previous comment you can remove the line 83, 84 and 85

Reply:
done. Thank you!!